### PR TITLE
Paper JOSS

### PR DESCRIPTION
**Pull request by YoungFaithful**
I'm looking forward for your feedback especially on the `paper.md` file.

### CODE REVIEW

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+``CapacityExpansion`` provides simple integration of (clustered) time-series, geographical, cost, and technology input data. The software features a modular model setup and an investment and dispatch optimization that uses the JUMP modeling language [@Dunning:2017]. An interface is provided between the optimization result and further analysis.
+
+# Infrastructure planning in the energy sector
+
+Energy systems convert different energy resources to meet desired demands like electric and thermal energy demands. Political, economical, and technological changes require changing the infrastructure of energy systems. Changing the infrastructure has to balance multiple political, environmental, and economical objectives. Capacity expansion planning can be an important tool during the planning process [@Gacitua:2018].
```
Comment (holgerteichgraeber) [inline paper/paper.md]:
```suggestion
Energy systems convert different energy resources to meet desired demands like electric and thermal energy demands. Political, economical, and technological changes require expansions of the infrastructure of energy systems. Expanding the infrastructure has to balance multiple political, environmental, and economical objectives. Capacity expansion planning can be an important tool during the planning process [@Gacitua:2018].
```

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Capacity expansion planning is used to compute cost-optimal energy system designs under given sets of constraints. The resulting cost-optimal energy system design can be used to inform policy decisions that incentivize the industry to invest in this design [@Johnston:2013]. Similarly, cost-optimal energy system designs can be used by companies for their investment strategies.
```
Comment (holgerteichgraeber) [inline paper/paper.md]:
```suggestion
Capacity expansion planning is used to compute cost-optimal energy system designs under given sets of constraints from the perspective of a central planner. The resulting cost-optimal energy system design can be used to inform policy decisions that incentivize the industry to invest in this design [@Johnston:2013]. Similarly, cost-optimal energy system designs can be used by companies for their investment strategies.
```

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+Capacity expansion planning is formulated as mathematical optimization problem. Like any optimization problem, capacity expansion planning has certain degrees of freedom, consists of constraints, and consists of an objective function that is minimized: Typical degrees of freedom (sometimes called decision variables) are installed capacities, power generation, energy storage, and power flows. We divide constraints into internal and external constraints: The internal constraints ensure that the model is physically consistent in itself and following the rules of thermodynamics, e.g. energy balances ensure energy conservation over time. External constraints restrict the solution space to external conditions like costs, demands, available energy resources as well as other political, environmental, economical, or technological constraints. The objective function to determine cost-optimal investment can be total system cost, the net present value, or another business-oriented cost measure.
```
Comment (holgerteichgraeber) [inline paper/paper.md]:
```suggestion
Capacity expansion planning is formulated as a mathematical optimization problem. Like any optimization problem, capacity expansion planning has certain degrees of freedom, consists of constraints, and consists of an objective function that is minimized: Typical degrees of freedom (also called decision variables) are installed capacities, power generation, energy storage, and power flows. We divide constraints into internal and external constraints: The internal constraints ensure that the model is physically consistent in itself and following the rules of thermodynamics, e.g. energy balances ensure energy conservation over time. External constraints restrict the solution space to external conditions like costs, demands, available energy resources as well as other political, environmental, economical, or technological constraints. The objective function to determine cost-optimal investment can be total system cost, the net present value, or another business-oriented cost measure.
```

Comment (holgerteichgraeber) [reply paper/paper.md]: Is the distinction between internal and external necessary? I find it slightly confusing, maybe take the distinction out and just list constraints.

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+Dispatch planning formulations are very similar to capacity expansion formulations. However, the installed capacity is no degree of freedom, but introduced as an external constraint.
+
+# Package features
+
+The model class of ``CapacityExpansion`` is capacity expansion planning and generation and transmission capacity expansion planning are combined. The model is setup as a linear optimization model that models energy systems based on the provided input data. Multiple energy carriers can be modeled, which makes the software suitable for research of sector coupling technologies. Technologies can be defined that belong to dispathable or non-dispathable generation, conversion, storage, transmission, or demand. The decision variables of the model are investment and dispatch and the total system costs are minimized.
```
Comment (holgerteichgraeber) [inline paper/paper.md]: Something is missing in the first sentence, gramatically.

File: paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+The following key features are provided by ``CapacityExpansion``. The usage and mathematical formulation is explained in detail within the software's documentation.
+
+- *Modeling language*: The implementation in Julia enables the usage of a high-level programming language, and short times for data handling at the same time. The implementation in Julia also allows the usage of the mathematical optimization language JUMP, which enables the optimization formulation close to mathematical writing. The integration into the Julia ecosystem further allows to handle the process from data preparation to figure export within one programming language and software.
```
Comment (holgerteichgraeber) [inline paper/paper.md]:
```suggestion
- *Modeling language*: The implementation in Julia enables the usage of a high-level programming language, and short times for data handling at the same time. The implementation in Julia also allows the usage of the mathematical optimization language JUMP [CITE JUMP], which enables the optimization formulation close to mathematical writing. The integration into the Julia ecosystem further allows to handle the process from data preparation to figure export within one programming language and software.
```

File: paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+- *The generalized import of input data*: Modeling other energy systems is possible by adjusting the input data. The package extracts all information needed to model a specific energy system based on a few standardized input files. The time-series, geographic, and cost data can be edited as tables and integrated using the `.csv`-file format. The technology input data can be edited like a tree structure and integrated using the `.yml`-file format.
+
+- *Integration of ``TimeSeriesClustering``*: Time-series data like the demand, available solar factors, and available wind factors are used as an input to model the temporal variance of the energy system. Aggregating the time-series input data is commonly done to reduce the computational complexity of the optimization. ``CapacityExpansion`` is well integrated with the Julia package ``TimeSeriesClustering``. This integration allows using the typical time-series aggregation methods, integrated testing of the temporal resolution, and an integrated feedback loops between the optimization result and time-series aggregation.
```
Comment (holgerteichgraeber) [inline paper/paper.md]:
```suggestion
- *Integration of ``TimeSeriesClustering``*: Time-series data like the demand, available solar factors, and available wind factors are used as an input to model the temporal variance of the energy system. Aggregating the time-series input data is commonly done to reduce the computational complexity of the optimization. ``CapacityExpansion`` is well integrated with the Julia package ``TimeSeriesClustering`` [CITE JOSS PAPER TIMESERIESCLUSTERING]. This integration allows using the typical time-series aggregation methods, integrated testing of the temporal resolution, and an integrated feedback loops between the optimization result and time-series aggregation.
```

File: paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+- *Seasonal storage*: A recent seasonal storage formulation from Kotzur is implemented to allow time-series aggregation and modeling of seasonal storage at the same time [@Kotzur:2018].
+
+- *Modular model setup*: The model setup is flexible and based on a modular setup. Depending on the configuration different technology groups can be activated or deactivated, a green or brown field study performed, emissions limits can be enforced, different storage models can be used, and an integrated investment and dispatch or pure dispatch optimization can be run.
+
+# ``CapacityExpansion`` within the broader ecosystem
+``CapacityExpansion`` is the first package to provide capacity expansion planning in Julia [@Bezanson:2017].
```
Comment (holgerteichgraeber) [inline paper/paper.md]: I would take out the reference to Bezanson here and add it further up. Bezanson doesnt talk about CEP, so may be confusing

File: paper/paper.md (hunk 6 above)
Comment (holgerteichgraeber) [reply paper/paper.md]: Also, please add citation to Applied Energy clustering paper after `typical time-series aggregation methods`.

**Review by holgerteichgraeber — CHANGES_REQUESTED** [review]
Nice paper and great additions to the documentation. Some comments in the code. 

You can now precompile the paper to PDF: https://whedon.theoj.org/. Please check how the references look, it seems to me that you have lots of information copied from Mendeley that you may not want printed in the paper. If it occurs, delete from .bib file. See my TimerSeriesClustering JOSS paper for the bibtex tags I included.

File: paper/paper.md (hunk 1 above)
Comment (YoungFaithful) [reply paper/paper.md]: I see the point. ;)

File: paper/paper.md (hunk 2 above)
Comment (YoungFaithful) [reply paper/paper.md]: Great addition

File: paper/paper.md (hunk 3 above)
Comment (YoungFaithful) [reply paper/paper.md]: I agree, that it is not necessary.

File: paper/paper.md (hunk 7 above)
Comment (YoungFaithful) [reply paper/paper.md]: I already had one reference at the beginning of the text and I'll take this one out.

File: paper/paper.md (hunk 6 above)
Comment (YoungFaithful) [reply paper/paper.md]: Added the citations.

File: paper/paper.md (hunk 5 above)
Comment (YoungFaithful) [reply paper/paper.md]: Added

File: paper/paper.md (hunk 4 above)
Comment (YoungFaithful) [reply paper/paper.md]: Would:
`The model class of ``CapacityExpansion`` is capacity expansion planning and it combines generation and transmission capacity expansion planning. ` make more sense?

**Pull request closed without merging**